File: easy/array/two_sum/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numberToIndex = {}
        for i, n in enumerate(nums):
            partner = target - n
            if partner in numberToIndex:
                return [numberToIndex[partner], i]
            numberToIndex[n] = i
        return


s = Solution()
print("Solution 1 : ", s.twoSum([2, 7, 11, 15], 9))
print("Solution 2 : ", s.twoSum([3, 3], 6))

File: easy/array/two_sum/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	numberToIndex := make(map[int]int, len(nums))
	for i, n := range nums {
		partner := target - n
		if partnerIndex, ok := numberToIndex[partner]; ok {
			return []int{partnerIndex, i}
		}

		numberToIndex[n] = i
	}
	return nil
}

func main() {
	fmt.Println("Solution 1 : ", twoSum([]int{2, 7, 11, 15}, 9))
	fmt.Println("Solution 2 : ", twoSum([]int{3, 3}, 6))
}
